specs/eventing/test-plan/error-signalling-conformance.md

```diff
@@ -0,0 +1,470 @@
+     status to another. We use VolatileTime in place of metav1.Time to exclude
+     this from creating equality.Semantic differences (all other things held
+     constant).
+
+   message      <string>
+     A human readable message indicating details about the transition.
+
+   reason       <string>
+     The reason for the condition's last transition.
+
+   severity     <string>
+     Severity with which to treat failures of this type of condition. When this
+     is not specified, it defaults to Error.
+
+   status       <string> -required-
+     Status of the condition, one of True, False, Unknown.
+
+   type <string> -required-
+     Type of condition.
+```
+
+---
+
+## [Test] OpenAPI schema for the Subscription kind
+
+Describe the registered `subscriptions.messaging.knative.dev` kind and verify
+that it contains a `status.conditions` field, expressed as a list of objects
+which themselves exclusively contain the fields:
+
+- `type`, typed as a string and marked as required
+- `status`, typed as a string
+- `message` typed as a string
+- `severity`, typed as a string
+- `lastTransitionTime`, typed as a string
+
+Explain the `status` and `status.conditions` fields:
+
+```
+kubectl explain subscriptions.messaging.knative.dev.status
+```
+
+```
+kubectl explain subscriptions.messaging.knative.dev.status.conditions
+```
+
+### [Output]
+
+```
+KIND:     Subscription
+VERSION:  messaging.knative.dev/v1
+
+RESOURCE: status <Object>
+
+[...]
+
+FIELDS:
+   [...]
+
+   conditions   <[]Object>
+     Conditions the latest available observations of a resource's current state.
+
+   [...]
+```
+
+```
+KIND:     Subscription
+VERSION:  messaging.knative.dev/v1
+
+RESOURCE: conditions <[]Object>
+
+[...]
+
+FIELDS:
+   lastTransitionTime   <string>
+     LastTransitionTime is the last time the condition transitioned from one
+     status to another. We use VolatileTime in place of metav1.Time to exclude
+     this from creating equality.Semantic differences (all other things held
+     constant).
+
+   message      <string>
+     A human readable message indicating details about the transition.
+
+   reason       <string>
+     The reason for the condition's last transition.
+
+   severity     <string>
+     Severity with which to treat failures of this type of condition. When this
+     is not specified, it defaults to Error.
+
+   status       <string> -required-
+     Status of the condition, one of True, False, Unknown.
+
+   type <string> -required-
+     Type of condition.
+```
+
+---
+
+## [Pre] Create a Broker instance which reconciliation fails
+
+```
+kubectl create -f specs/eventing/test-plan/control-plane/error-signalling/broker-not-ready.yaml
+```
+
+## [Test] Broker failure is propagated to the Ready status condition
+
+The failure caused by the intentionally invalid spec is propagated to the `Ready` condition, which has a `status` of
+`False`, a severity of `""` (error), a non-empty `reason`, and a non-empty `message`:
```

Comment:
I think the message does not need to be non empty to be conformant (although I agree that it is expected to have such message informed)

```
Conditions with a status other than "True" SHOULD provide message and reason fields indicating the reason that the status is not "True". Conditions where the status is "False" MUST provide a failure reason in the condition. ("Unknown" conditions might not have been reconciled, and so MAY have an empty reason.)
```

Reply:
You're right. We only need to validate that a _reason_ is provided.

Reply:
Done in [8386464e..a83ce0c7](https://github.com/knative/specs/compare/8386464ee10e5b061eaee0ce840bdf92d9d9bbcd..a83ce0c7c7b46214fd4e069690c785cc3c76c1dc)